### ISSUE TITLE
Metadata directive "REPLACE" erroneously clears file contents when copying file to its existing path

### ISSUE THREAD

**Issue by kjmahalingam** [adobe/S3Mock#569]
With a local S3Mock instance running, I uploaded a file `test.txt`. Its content length is 16, as seen below.
```
$ aws --endpoint-url=http://localhost:9091 --no-verify-ssl s3 cp \
test.txt \
"s3://test-bucket/test.txt"

upload: ./test.txt to s3://test-bucket/test.txt

$ aws --endpoint-url=http://localhost:9091 --no-verify-ssl s3 ls test-bucket
2022-04-29 11:35:17         16 test.txt
```

When copying this file to a new destination and replacing its metadata, we can see that the file is copied correctly.
```
$ aws --endpoint-url=http://localhost:9091 --no-verify-ssl s3 cp \
"s3://test-bucket/test.txt" \
"s3://test-bucket/test_copy.txt" \
--metadata-directive "REPLACE" --metadata "foo=bar"
copy: s3://test-bucket/test.txt to s3://test-bucket/test_copy.txt

$ aws --endpoint-url=http://localhost:9091 --no-verify-ssl s3 ls test-bucket
2022-04-29 11:35:17         16 test.txt
2022-04-29 11:38:04         16 test_copy.txt
```

However, when copying this file to _its existing location_ and replacing its metadata, we can see that the file's content length is now 0, meaning the contents of the file have been deleted.
```
$ aws --endpoint-url=http://localhost:9091 --no-verify-ssl s3 cp \
"s3://test-bucket/test.txt" \
"s3://test-bucket/test.txt" \
--metadata-directive "REPLACE" --metadata "foo=bar"
copy: s3://test-bucket/test.txt to s3://test-bucket/test.txt

$ aws --endpoint-url=http://localhost:9091 --no-verify-ssl s3 ls test-bucket
2022-04-29 11:38:37          0 test.txt
2022-04-29 11:38:04         16 test_copy.txt
```

We can see that both files do have the correct updated metadata, but that the `ContentLength` attribute of the one that was copied into its original location is now 0.
```
$ aws --endpoint-url=http://localhost:9091 --no-verify-ssl s3api head-object --bucket "test-bucket" --key "test.txt"
{
    "LastModified": "Fri, 29 Apr 2022 15:38:37 GMT",
    "ContentLength": 0,
    "ETag": "\"d41d8cd98f00b204e9800998ecf8427e\"",
    "ContentType": "text/plain",
    "Metadata": {
        "foo": "bar"
    }
}

$ aws --endpoint-url=http://localhost:9091 --no-verify-ssl s3api head-object --bucket "test-bucket" --key "test_copy.txt"
{
    "LastModified": "Fri, 29 Apr 2022 15:38:04 GMT",
    "ContentLength": 16,
    "ETag": "\"a28bca1b906f539ba70ca3a0b1f2e773\"",
    "ContentType": "text/plain",
    "Metadata": {
        "foo": "bar"
    }
}
```

When performing this same operation in a live S3 environment, this behavior does not occur, and the file that is copied to its existing location does still retain its existing contents.

**Comment by afranken** [comment]
Thank you for the report - this was fixed with #468 but I did not release after merging.

I will release a new version today :)

**Comment by afranken** [comment]
just noticed that we are actually not replacing the metadata in this case.
So only copyObject with "COPY" is working correctly.

I'll fix this and then make a release. Probably tomorrow.

**Comment by kjmahalingam** [comment]
Thank you for looking into this! Please let me know if you need me to try anything else out locally.

**Comment by afranken** [comment]
I just released 2.4.10 including my fix :)
